### DISCUSSION TITLE
ErrorBoundaries fail to catch errors produced by atomWithQuery when using Provider for apps rendered by ReactDOM.createRoot

### DISCUSSION THREAD

**Discussion by matthewconstantine** [pmndrs/jotai#1344] [Q&A] [answered]
We're running into an issue with `atomWithQuery` and `Provider`. 

### Expected:
When a `atomWithQuery` throws an error, it should reach the `ErrorBoundary`. 

### Actual:
The error thrown never reaches the boundary. Instead, `atomWithQuery` throws an error `Error: setting result without mount`. That error also does not reach the `ErrorBoundary`. We have `Suspense` wrapper, but the app remains suspended indefinitely. 

This was a tricky one to narrow down. It only happens when components are wrapped within a jotai `<Provider>`. And it only happens for apps rendered with `ReactDOM.createRoot` which is required for React 18. Using `react-dom` `render` does not result in this behavior, however React complains with a deprecation alert saying:

> "ReactDOM.render is no longer supported in React 18. Use createRoot instead. Until you switch to the new API, your app will behave as if it's running React 17. 

Example sandbox: https://codesandbox.io/s/loving-field-9zp8ii?file=/src/App.tsx

Are there any workarounds we can try?

**Comment by dai-shi** [comment, marked as answer]
https://codesandbox.io/s/hopeful-cherry-wc64wk?file=/src/App.tsx

I'm not sure if we state it clearly in docs, but we should have:
```jsx
<Provider>
  <ErrorBoundary>
    <Suspense>
```
in this order.
